src/VirtualClient/VirtualClient.Dependencies/AMDGPUDriverInstallation.cs

```diff
@@ -0,0 +1,139 @@
+                    default:
+                        return this.Parameters.GetValue<bool>(nameof(AMDGPUDriverInstallation.RebootRequired), true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// A policy that defines how the component will retry when
+        /// it experiences transient issues.
+        /// </summary>
+        public IAsyncPolicy RetryPolicy { get; set; }
+
+        /// <summary>
+        /// Executes  GPU driver installation steps.
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.ExecutionStarted", telemetryContext);
+            State installationState = await this.stateManager.GetStateAsync<State>(nameof(AMDGPUDriverInstallation), cancellationToken)
+                .ConfigureAwait(false);
+            if (installationState == null)
+            {
+                if (this.Platform == PlatformID.Win32NT)
+                {
+                    await this.AMDDriverInstallationOnWindowsAsync(telemetryContext, cancellationToken)
+                               .ConfigureAwait(false);
+                    await this.stateManager.SaveStateAsync(nameof(this.AMDDriverInstallationOnWindowsAsync), new State(), cancellationToken)
+                        .ConfigureAwait(false);
+                }
+
+                VirtualClientRuntime.IsRebootRequested = this.RebootRequired;
+            }
+
+            this.Logger.LogTraceMessage($"{this.TypeName}.ExecutionCompleted", telemetryContext);
+        }
+
+        private async Task AMDDriverInstallationOnWindowsAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string installerPath = string.Empty;
+            DependencyPath amdDriverInstallerPackage = await this.packageManager.GetPackageAsync(
+                this.PackageName, cancellationToken)
+                    .ConfigureAwait(false);
+            if (this.fileSystem.Directory.GetFiles(amdDriverInstallerPackage.Path, "*.exe", SearchOption.AllDirectories).Length > 0)
+            {
+                installerPath = this.fileSystem.Directory.GetFiles(amdDriverInstallerPackage.Path, "*.exe", SearchOption.AllDirectories)[0];
+            }
+            else
+            {
+                throw new DependencyException($"The installer file was not found in the directory {amdDriverInstallerPackage.Path}", ErrorReason.DependencyNotFound);
+            }
+
+            await this.ExecuteCommandAsync(installerPath, "/S /v/qn", Environment.CurrentDirectory, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        private Task ExecuteCommandAsync(string commandLine, string commandLineArgs, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken)
```

Comment:
We already have executecommandasync in virtualclientcomponent extensions that can cover portion of this function